### PR TITLE
Feed chart lttb downsampling

### PR DESCRIPTION
**Pull request by ncarazon**
### Summary

Related to #4937

Adds feed-only chart data reduction to reduce the amount of chart data rendered in feed previews while keeping full-resolution charts on question pages.

Implemented changes:

Added a local LTTB utility with tests for endpoint preservation, spike preservation, target sizing, and non-numeric/null handling. 

Applied LTTB only to feed continuous distribution curves: 

* MinifiedContinuousAreaChart 
* ContinuousAreaChart  

Added a boundary-preserving step reducer for feed timeline charts where LTTB could shift transition timestamps: 

* NumericTimeline 
* GroupChart 

Skipped discrete distributions, FanChart, bar-style TimeSeriesChart, and MultipleChoiceChart because their data shape or rendering model does not fit LTTB safely.

Performance notes:

* With Chrome CPU throttling set to `6x slowdown`, initial render improved from \~13s before to \~9s after in local testing.
* This should also provide a small rendering boost while scrolling through the virtualized feed, since eligible newly mounted feed charts have fewer points for Victory to process and render.

## Summary by CodeRabbit

* **New Features**
  * Continuous area charts now support a display mode to optimize feed vs. question rendering.
  * Feed charts downsample dense trend lines and apply step-segment reduction to keep visuals readable.
  * Multiple-choice feed timelines preserve step alignment during downsampling.

* **Bug Fixes**
  * Improved downsampling and step reduction for series containing missing values while retaining important endpoints/peaks.
  * Chart data now updates correctly when the rendering mode changes.

* **Tests**
  * Expanded coverage for downsampling and step-reduction behaviors, including edge cases.

### CODE REVIEW

File: front_end/src/utils/charts/lttb.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+): Line<X, Y> {
+  if (targetCount < 3 || targetCount >= points.length) {
+    return points;
+  }
+
+  const numericPoints: NumericPoint[] = [];
+  for (let i = 0; i < points.length; i++) {
+    const point = points[i] as LinePoint<X, Y>;
+    const x = Number(point.x);
+    const y = Number(point.y);
```
Comment (cemreinanc) [inline front_end/src/utils/charts/lttb.ts:27]: Im not sure if it's a reachable path but tiny edge case here: `Number(null)` is `0`, so null `y` values still make it into `numericPoints` as real samples. That means a null can sometimes affect bucket averages or even be returned, despite the comment/test saying non-numeric/null values are skipped. Could we explicitly skip these nulls before coercing? I'd also tweak the test so it fails on the current behavior, since the existing target count happens not to select the nulls.

**Review by cemreinanc — COMMENTED** [review]
I traced the forecaster-feed multiple-choice path, and it still renders through `MultipleChoiceChart` with the full CP `line` arrays passed straight into `VictoryArea`. So this path is not using the new downsampled / boundary-safe data. Consumer feed uses a separate percentage-bar path, so this is specific to forecaster non-group MC cards.

File: front_end/src/utils/charts/lttb.ts (hunk 1 above)
Comment (ncarazon) [reply front_end/src/utils/charts/lttb.ts:27]: > I traced the forecaster-feed multiple-choice path, and it still renders through MultipleChoiceChart with the full CP line arrays passed straight into VictoryArea. So this path is not using the new downsampled / boundary-safe data. Consumer feed uses a separate percentage-bar path, so this is specific to forecaster non-group MC cards.

I added feed-only reduction there using a shared boundary-preserving mask across active options, so VictoryStack stays x-aligned and stepAfter transitions are preserved. Added test coverage for it too.

> Im not sure if it's a reachable path but tiny edge case here: Number(null) is 0, so null y values still make it into numericPoints as real samples. That means a null can sometimes affect bucket averages or even be returned, despite the comment/test saying non-numeric/null values are skipped. Could we explicitly skip these nulls before coercing? I'd also tweak the test so it fails on the current behavior, since the existing target count happens not to select the nulls.

Fixed. We now explicitly skip null x/y values before numeric coercion, so null gap markers cannot become real y=0 samples.

**Review by cemreinanc — APPROVED** [review]
LGTM
